Skip post-deployment notifications when empty package.xml, simplify JIRA error logging

legetz opened the pull request:
Skip notification providers if there was nothing to deploy (in delta mode)

nvuillam commented on src/commands/hardis/project/deploy/sources/dx.ts:
```diff
@@ -387,13 +387,15 @@ If you need to increase the deployment waiting time (force:source:deploy --wait
+
     // Set ListViews to scope Mine if defined in .sfdx-hardis.yml
-    if (this.configInfo.listViewsToSetToMine && this.checkOnly === false) {
+    if (this.configInfo.listViewsToSetToMine && deployExecuted) {
```
@legetz why removing the test on checkonly ? it can be equal to true with deployExecuted > 0 if we simulate a deployment during a PR check ? :)

legetz replied:
@nvuillam : I added new const, deployExecuted which basically means true deployment executed:
`const deployExecuted = !this.checkOnly && deployXmlCount > 0 ? true : false;`
So in the case of simulation/check, deployExecuted will never be true.

Since that already takes into account `checkOnly`, I removed that from this line. Makes sense?

nvuillam replied:
You're right, i didn't look close enough ;)

nvuillam commented on src/commands/hardis/project/deploy/sources/dx.ts:
```diff
@@ -387,13 +387,15 @@ If you need to increase the deployment waiting time (force:source:deploy --wait
 
     // Send notification of deployment success
-    if (!this.checkOnly) {
+    if (deployExecuted) {
```
@legetz why removing the test on checkonly ? it can be equal to true with deployExecuted > 0 if we simulate a deployment during a PR check ? :)

legetz replied:
Same logic here than above

nvuillam reviewed (changes requested):
Plz check comments :)

legetz commented:
Added another improvement to simplify/prettify JIRA error logging.

Old output:
```
[sfdx-hardis] [JiraProvider] Try to post comments on 1 tickets...
[sfdx-hardis] [JiraProvider] Error while posting comment on ABC-826
This is a probably a config/rights errors as the response contain HTML
[sfdx-hardis] [JiraProvider] Error while adding label DEV_DEPLOYED on ABC-826
"<!doctype html><html lang=\"en\"><head><title>HTTP Status 405 – Method Not Allowed</title><style type=\"text/css\">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 405 – Method Not Allowed</h1><hr class=\"line\" /><p><b>Type</b> Status Report</p><p><b>Description</b> The method received in the request-line is known by the origin server but not supported by the target resource.</p><hr class=\"line\" /><h3>Apache Tomcat/9.0.80</h3></body></html>"
Error: "<!doctype html><html lang=\"en\"><head><title>HTTP Status 405 – Method Not Allowed</title><style type=\"text/css\">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 405 – Method Not Allowed</h1><hr class=\"line\" /><p><b>Type</b> Status Report</p><p><b>Description</b> The method received in the request-line is known by the origin server but not supported by the target resource.</p><hr class=\"line\" /><h3>Apache Tomcat/9.0.80</h3></body></html>"
    at JiraApi.doRequest (/root/.local/share/sf/node_modules/jira-client/lib/jira.js:347:13)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async JiraProvider.postDeploymentComments (/root/.local/share/sf/node_modules/sfdx-hardis/lib/common/ticketProvider/jiraProvider.js:171:21)
    at async TicketProvider.postDeploymentActions (/root/.local/share/sf/node_modules/sfdx-hardis/lib/common/ticketProvider/index.js:50:17)
    at async DxSources.run (/root/.local/share/sf/node_modules/sfdx-hardis/lib/commands/hardis/project/deploy/sources/dx.js:191:17)
    at async DxSources._run (/root/.local/share/sf/node_modules/sfdx-hardis/node_modules/@salesforce/command/lib/sfdxCommand.js:81:40)
    at async Config.runCommand (/usr/local/lib/node_modules/@salesforce/cli/node_modules/@oclif/core/lib/config/config.js:417:25)
    at async run (/usr/local/lib/node_modules/@salesforce/cli/node_modules/@oclif/core/lib/main.js:85:16)
[sfdx-hardis] [JiraProvider] Posted comments on 0 ticket(s): 
[sfdx-hardis] [JiraProvider] Added label DEV_DEPLOYED on 0 ticket(s): 
```

Simplified/new output:
```
[sfdx-hardis] [JiraProvider] Try to post comments on 1 tickets...
[sfdx-hardis] [JiraProvider] Error while posting comment on ABC-826: Probably config/access error since response is HTML
[sfdx-hardis] [JiraProvider] Error while adding label DEV_DEPLOYED on ABC-826: Probably config/access error since response is HTML
```

nvuillam reviewed (approved):
Great, thanks for this contribution @legetz :)

nvuillam commented:
@legetz next time please think to also add entries in CHANGELOG.md :)
